### PR TITLE
fix watch issue

### LINKED ISSUE #2616: Integration and verification of third-party CNI flannel based on the edge side list-watch
**What happened**:
flannel cant watch nodes
**What you expected to happen**:
flannel run with no error
**How to reproduce it (as minimally and precisely as possible)**:
use flannel.yaml need to  add args `--kube-api-url=http://127.0.0.1:10550`
**Anything else we need to know?**:
````
# flannel  log
flannel pod use arg --v=10 to get detail log:

I0310 05:57:46.580584       1 round_trippers.go:398] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: flanneld/v1.6.10 (linux/amd64) kubernetes/$Format" http://127.0.0.1:10550/api/v1/nodes?resourceVersion=12402179&timeoutSeconds=593&watch=true
I0310 05:57:46.583085       1 round_trippers.go:417] GET http://127.0.0.1:10550/api/v1/nodes?resourceVersion=12402179&timeoutSeconds=593&watch=true 200 OK in 2 milliseconds
I0310 05:57:46.583114       1 round_trippers.go:423] Response Headers:
I0310 05:57:46.583122       1 round_trippers.go:426]     Content-Type: application/json
I0310 05:57:46.583129       1 round_trippers.go:426]     Date: Wed, 10 Mar 2021 05:57:46 GMT
E0310 05:57:46.583926       1 streamwatcher.go:109] Unable to decode an event from the watch stream: unable to decode watch event: Object 'Kind' is missing in '{"metadata":{},"status":"Failure","message":"Internal error occurred: the server could not find the requested resource","reason":"InternalError","details":{"causes":[{"message":"the server could not find the requested resource"}]},"code":500}'
I0310 05:57:46.584046       1 reflector.go:402] github.com/coreos/flannel/subnet/kube/kube.go:307: Unexpected watch close - watch lasted less than a second and no items received
W0310 05:57:46.584083       1 reflector.go:323] github.com/coreos/flannel/subnet/kube/kube.go:307: watch of *v1.Node ended with: very short watch

the error is show because:
curl http://127.0.0.1:10550/api/v1/nodes?watch=true return 500 error


# edgecore log
3月 10 14:01:47 localhost.localdomain edgecore[2935152]: I0310 14:01:47.211996 2935152 storage.go:159] [metaserver/reststorage] successfully apply for a watch listener (/api/v1/nodes) through cloud
3月 10 14:01:47 localhost.localdomain edgecore[2935152]: I0310 14:01:47.212292 2935152 watcher.go:212] start watching, rev:0
3月 10 14:01:47 localhost.localdomain edgecore[2935152]: I0310 14:01:47.212329 2935152 imitator.go:141] /v1, Resource=nodes,,
3月 10 14:01:47 localhost.localdomain edgecore[2935152]: E0310 14:01:47.212550 2935152 watcher.go:215] failed to sync with latest state: the server could not find the requested resource

# cloudcore log

3月 10 14:02:47 fp-2 cloudcore[53768]: W0310 14:02:47.696235   53768 listener.go:68] built message resource failed with error: required parameter are not set (node id, namespace or resource type)


````
**Environment**:
- Kubernetes version (use `kubectl version`):
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`):
  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>

### PR DESCRIPTION
**Pull request by Abirdcfly**
Signed-off-by: Abirdcfly <fp544037857@gmail.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #2616 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cloud/pkg/dynamiccontroller/application/listener.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package application
 
 import (
 	"github.com/google/uuid"
+	v2 "github.com/kubeedge/kubeedge/edge/pkg/metamanager/dao/v2"
```
Comment (fisherxu) [inline cloud/pkg/dynamiccontroller/application/listener.go:5]: Need sort the imports :)

File: cloud/pkg/dynamiccontroller/application/listener.go [hunk 2]
```diff
@@ -52,7 +53,15 @@ func (l *SelectorListener) sendObj(event watch.Event, messageLayer messagelayer.
...
 	msg.SetResourceVersion(accessor.GetResourceVersion())
-	resource, err := messagelayer.BuildResource(l.nodeName, accessor.GetNamespace(), l.gvr.Resource, accessor.GetName())
+	name := accessor.GetName()
+	if name == "" {
+		name = v2.NullName
+	}
+	namespace := accessor.GetNamespace()
+	if namespace == "" {
+		namespace = v2.NullNamespace
+	}
```
Comment (fisherxu) [inline cloud/pkg/dynamiccontroller/application/listener.go:63]: Looks good, the namespace can be null for clusterscope object, but name may impossible to be null? :)

**Review by fisherxu — COMMENTED** [review]
/lgtm

File: cloud/pkg/dynamiccontroller/application/listener.go (hunk 2 above)
Comment (fisherxu) [reply cloud/pkg/dynamiccontroller/application/listener.go:63]: With this patch, flannel can run in the edge?

Comment (Abirdcfly) [reply cloud/pkg/dynamiccontroller/application/listener.go:59]: You are right. name not be null. And with this patch.flannel should be run in edge.

Comment (fisherxu) [reply cloud/pkg/dynamiccontroller/application/listener.go:59]: > And with this patch.flannel should be run in edge.

Wow!! Great! 

> name not be null.

But I think we can keep the name related logic here?

**Comment by fisherxu** [comment]
Could you please cherry-pick this PR to release-1.6 using `hack/cherry_pick_pull.sh` script? @Abirdcfly  Thanks.

File: cloud/pkg/dynamiccontroller/application/listener.go (hunk 2 above)
Comment (Abirdcfly) [reply cloud/pkg/dynamiccontroller/application/listener.go:59]: > But I think we can keep the name related logic here?

Both are ok.(◐‿◑)﻿

**Pull request merged**